src/Components/Checkbox.tsx

```diff
@@ -1,71 +1,112 @@
-import React from "react"
-
+import React, { Component, HTMLProps } from "react"
 import styled from "styled-components"
 import colors from "../Assets/Colors"
 
-interface Props extends React.HTMLProps<Checkbox> {
+export interface CheckboxState {
   checked: boolean
 }
 
-export class Checkbox extends React.Component<Props, null> {
-  static defaultProps = {
-    checked: true,
+export class Checkbox extends Component<HTMLProps<Checkbox>, CheckboxState> {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      checked: props.checked || false,
+    }
+  }
+
+  onChange = event => {
+    this.props.onChange && this.props.onChange(event)
```

Comment:
While this is totally valid and looks clean I think the literal version would be more maintainable long-term:

```javascript
if (this.props.onChange) {
  this.props.onChange(event)
}
```